app/controllers/trainees/subject_specialisms_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Trainees
+  class SubjectSpecialismsController < ApplicationController
+    before_action :authorize_trainee
+    before_action :load_subject_specialisms
+
+    def edit
+      @subject_specialisms_form = LanguageSubjectSpecialismsForm.new(trainee)
+    end
+
+    def update
+      @subject_specialisms_form = LanguageSubjectSpecialismsForm.new(trainee, params: subject_specialism_params, user: current_user)
+
+      save_strategy = trainee.draft? ? :save! : :stash
+
+      if @subject_specialisms_form.public_send(save_strategy)
+        # TODO: do something
+      else
+        render :edit
+      end
+    end
+
+  private
+
+    def trainee
+      @trainee ||= Trainee.from_param(params[:trainee_id])
+    end
+
+    def course
+      trainee.available_courses.find_by!(code: course_code)
+    end
+
+    def course_code
```

Comment:
Unsolved! We're currently unsure how these 'middle' forms should know about the publish course code, so currently I've hacked it into a query string parameter.

Potentially the first form `PublishCourseDetails` should save/stash the course_code onto the trainee/data model and we can pull it from there?

Then the confirm details form can more-or-less behave like the rest of confirm forms.